### PR TITLE
GEODE-7992: Implement a test class for the EXISTS Redis API command

### PR DESCRIPTION
**Pull request by sabbeyPivotal**
The Redis EXISTS command verifies that a key exists (https://redis.io/commands/exists).  The EXISTS command was previously implemented, but not tested. This PR adds tests and fixes implementation for this command.

### CODE REVIEW

File: geode-redis/src/integrationTest/java/org/apache/geode/redis/general/PersistIntegrationTest.java [hunk 1]
```diff
@@ -130,7 +130,7 @@ public void shouldPersistKey_givenKeyWith_geoValue() {
     String geoKey = "sicily";
     double latitude = 13.361389;
     double longitude = 38.115556;
-    String geoMember = "Palermo Catina";
```
Comment (ringles) [inline geode-redis/src/integrationTest/java/org/apache/geode/redis/general/PersistIntegrationTest.java:133]: Precision is important. :)

File: geode-redis/src/integrationTest/java/org/apache/geode/redis/general/ExistsIntegrationTest.java [hunk 2]
```diff
@@ -0,0 +1,348 @@
...
+    }
+  }
+
+  private void existsKeys(AtomicLong existsCount, Jedis jedis, int iterationCount) {
+    for (int i = 0; i < iterationCount; i++) {
+      existsCount.addAndGet(jedis.exists("key" + i, "doesNotExist" + i));
+    }
+  }
+
+  private class LoopingThread extends Thread {
```
Comment (ringles) [inline geode-redis/src/integrationTest/java/org/apache/geode/redis/general/ExistsIntegrationTest.java:325]: I don't think it's absolutely critical for this commit, but we probably need to standardize on an approach for (a) spawning and executing threads, and (b) iterated runnables. There's a few different approaches scattered across the tests.

File: geode-redis/src/main/java/org/apache/geode/redis/internal/RegionProvider.java [hunk 3]
```diff
@@ -210,7 +210,7 @@ public boolean removeKey(ByteArrayWrapper key, RedisDataType type, boolean cance
           return stringsRegion.remove(key) != null;
         } else if (type == RedisDataType.REDIS_HLL) {
           return hLLRegion.remove(key) != null;
-        } else if (type == RedisDataType.REDIS_LIST) {
+        } else if (type == RedisDataType.REDIS_LIST || type == RedisDataType.REDIS_SORTEDSET) {
```
Comment (ringles) [inline geode-redis/src/main/java/org/apache/geode/redis/internal/RegionProvider.java:213]: Do we have stories in the backlog about sorted sets? Looks like we missed this earlier.

File: geode-redis/src/distributedTest/java/org/apache/geode/redis/executors/ExistsDUnitTest.java [hunk 4]
```diff
@@ -0,0 +1,76 @@
...
+import org.apache.geode.test.awaitility.GeodeAwaitility;
+import org.apache.geode.test.dunit.rules.ClusterStartupRule;
+import org.apache.geode.test.dunit.rules.MemberVM;
+import org.apache.geode.test.junit.categories.RedisTest;
+
+@Category({RedisTest.class})
+public class ExistsDUnitTest implements Serializable {
+
+  @ClassRule
+  public static ClusterStartupRule cluster = new ClusterStartupRule(5);
```
Comment (ringles) [inline geode-redis/src/distributedTest/java/org/apache/geode/redis/executors/ExistsDUnitTest.java:38]: I think we're starting five VMs here, but only using three.

File: geode-redis/src/integrationTest/java/org/apache/geode/redis/general/ExistsIntegrationTest.java (hunk 2 above)
Comment (sabbeyPivotal) [reply geode-redis/src/integrationTest/java/org/apache/geode/redis/general/ExistsIntegrationTest.java:325]: I totally agree!  Let's have a techtro!

File: geode-redis/src/main/java/org/apache/geode/redis/internal/RegionProvider.java (hunk 3 above)
Comment (sabbeyPivotal) [reply geode-redis/src/main/java/org/apache/geode/redis/internal/RegionProvider.java:213]: Added a placeholder for sorted sets.

File: geode-redis/src/distributedTest/java/org/apache/geode/redis/executors/ExistsDUnitTest.java (hunk 4 above)
Comment (sabbeyPivotal) [reply geode-redis/src/distributedTest/java/org/apache/geode/redis/executors/ExistsDUnitTest.java]: We are.  Thank you for catching that.  I just updated it to 3.

File: geode-redis/src/integrationTest/java/org/apache/geode/redis/general/ExistsIntegrationTest.java [hunk 5]
```diff
@@ -0,0 +1,348 @@
...
+    return strings;
+  }
+
+  private void setKeys(Jedis jedis, int iterationCount) {
+    for (int i = 0; i < iterationCount; i++) {
+      jedis.set("key" + i, "value" + i);
+    }
+  }
+
+  private void existsKeys(AtomicLong existsCount, Jedis jedis, int iterationCount) {
```
Comment (dschneider-pivotal) [inline geode-redis/src/integrationTest/java/org/apache/geode/redis/general/ExistsIntegrationTest.java:319]: Is this unused? If so delete it

File: geode-redis/src/integrationTest/java/org/apache/geode/redis/general/ExistsIntegrationTest.java [hunk 6]
```diff
@@ -0,0 +1,348 @@
...
+    Thread loopingThread2 = new LoopingThread((i) -> jedis2.exists(toArray("key")), iterationCount);
+    Thread loopingThread3 = new LoopingThread((i) -> jedis3.del("key"), iterationCount);
+
+    loopingThread1.start();
+    loopingThread2.start();
+    loopingThread3.start();
+    loopingThread1.join();
+    loopingThread2.join();
+    loopingThread3.join();
+  }
```
Comment (dschneider-pivotal) [inline geode-redis/src/integrationTest/java/org/apache/geode/redis/general/ExistsIntegrationTest.java:307]: Nothing to assert? Are we just happy that none of the threads throws an exception? If so the method name should communicate this.

File: geode-redis/src/integrationTest/java/org/apache/geode/redis/general/ExistsIntegrationTest.java (hunk 5 above)
Comment (prettyClouds) [reply geode-redis/src/integrationTest/java/org/apache/geode/redis/general/ExistsIntegrationTest.java:319]: thanks, good catch.

File: geode-redis/src/integrationTest/java/org/apache/geode/redis/general/ExistsIntegrationTest.java (hunk 6 above)
Comment (prettyClouds) [reply geode-redis/src/integrationTest/java/org/apache/geode/redis/general/ExistsIntegrationTest.java:307]: updated, the name. We just wanted to make sure there were no Concurrent Modification Exceptions on any underlying data structures.  Not sure what else to test. thx

**Pull request merged**